File: packages/plugin/android/src/main/java/ca/zyra/capacitor/stripe/EphKeyProvider.java
package ca.zyra.capacitor.stripe;

import com.stripe.android.EphemeralKeyProvider;
import com.stripe.android.EphemeralKeyUpdateListener;

import org.jetbrains.annotations.NotNull;

class EphKeyProvider implements EphemeralKeyProvider {
    private static String key;

    public static void setKey(String key) {
        EphKeyProvider.key = key;
    }

    @Override
    public void createEphemeralKey(@NotNull String s, @NotNull EphemeralKeyUpdateListener ephemeralKeyUpdateListener) {
        ephemeralKeyUpdateListener.onKeyUpdate(EphKeyProvider.key);
    }
}

File: packages/plugin/android/src/main/java/ca/zyra/capacitor/stripe/EphKeyProvider.kt
package ca.zyra.capacitor.stripe

import com.stripe.android.EphemeralKeyProvider
import com.stripe.android.EphemeralKeyUpdateListener

internal class EphKeyProvider(internal val key: String) : EphemeralKeyProvider {
    override fun createEphemeralKey(apiVersion: String, keyUpdateListener: EphemeralKeyUpdateListener) {
        keyUpdateListener.onKeyUpdate(key)
    }
}
